From 342384fc38345a1f271673f38884746b8549b0f4 Mon Sep 17 00:00:00 2001
From: Simon Busch <simon.busch@lge.com>
Date: Fri, 21 Nov 2014 15:46:21 +0100
Subject: [PATCH] gdbus-codegen: replace all non alpha characters with an
 underscore

We need to replace all non alpha characters in the header guard name
to avoid breaking the build when the build path includes not allowed
characters.

Upstream-Status: pending
---
 gio/gdbus-2.0/codegen/codegen.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff -uNr glib-2.56.1.orig/gio/gdbus-2.0/codegen/codegen.py glib-2.56.1/gio/gdbus-2.0/codegen/codegen.py
--- glib-2.56.1.orig/gio/gdbus-2.0/codegen/codegen.py	2018-05-04 16:06:38.002594081 +0000
+++ glib-2.56.1/gio/gdbus-2.0/codegen/codegen.py	2018-05-04 19:21:46.989590963 +0000
@@ -58,7 +58,7 @@
         self.namespace, self.ns_upper, self.ns_lower = generate_namespace(namespace)
         self.generate_objmanager = generate_objmanager
         self.generate_autocleanup = generate_autocleanup
-        self.header_guard = header_name.upper().replace('.', '_').replace('-', '_').replace('/', '_').replace(':', '_')
+        self.header_guard = "".join([c if c.isalpha() else "_" for c in header_name.upper()])
         self.use_pragma = use_pragma
         self.outfile = outfile
 
